src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,861 @@
+     * @return the string without square brackets.
+     */
+    private static String removeSquareBrackets(String value) {
+        return value
+                .replace("[", "")
+                .replace("]", "");
+    }
+
+    /**
+     * Gets the name of the bean property's default value for the string array class.
+     *
+     * @param propertyName The bean property's name
+     * @param value The bean property's value
+     * @return String form of property's default value
+     */
+    private static String getStringArrayPropertyValue(String propertyName, Object value) {
+        String result;
+        if (value == null) {
+            result = "";
+        }
+        else {
+            final Stream<?> valuesStream;
+            if (value instanceof Collection) {
+                final Collection<?> collection = (Collection<?>) value;
+                valuesStream = collection.stream();
+            }
+            else {
+                final Object[] array = (Object[]) value;
+                valuesStream = Arrays.stream(array);
+            }
+            result = valuesStream
+                .map(String.class::cast)
+                .sorted()
+                .collect(Collectors.joining(COMMA_SPACE));
+            valuesStream.close();
+        }
+
+        if (result.isEmpty()) {
+            if (FILE_EXTENSIONS.equals(propertyName)) {
+                result = "all files";
+            }
+            else {
+                result = CURLY_BRACKETS;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns the name of the bean property's default value for the int array class.
+     *
+     * @param value The bean property's value.
+     * @return String form of property's default value.
+     */
+    private static String getIntArrayPropertyValue(Object value) {
+        final IntStream stream;
+        if (value instanceof Collection) {
+            final Collection<?> collection = (Collection<?>) value;
+            stream = collection.stream()
+                    .mapToInt(number -> (int) number);
+        }
+        else if (value instanceof BitSet) {
+            stream = ((BitSet) value).stream();
+        }
+        else {
+            stream = Arrays.stream((int[]) value);
+        }
+        String result = stream
+                .mapToObj(TokenUtil::getTokenName)
+                .sorted()
+                .collect(Collectors.joining(COMMA_SPACE));
+        if (result.isEmpty()) {
+            result = CURLY_BRACKETS;
+        }
+        stream.close();
+        return result;
+    }
+
+    /**
+     * Gets the class of the given field.
+     *
+     * @param field the field to get the class of.
+     * @return the class of the field.
+     * @throws MacroExecutionException if an error occurs during getting the class.
+     */
+    private static Class<?> getFieldClass(Field field) throws MacroExecutionException {
+        Class<?> result = null;
+
+        if (field != null) {
```

Comment:
hmm ..... in what cases field is null ?
soundlike this condition shuld be out of this methods, as if upper level do bad field calculation it should handle it itself, `getFiledClass(null)` sounds weird.

Reply:
`HeaderCheck.header` for example.

Reply:
https://checkstyle.org/checks/header/header.html#Properties

Field can not be null.
Value of field can be null. Do we have issue with method name?